lib/tools.js

```diff
@@ -1932,6 +1932,63 @@ function removeIdFromAllEnums(objects, id) {
     });
 }
 
+/**
+ * Validates types of obj.common properties and object.type, if invalid types are used, an error is thrown.
+ * If attributes of obj.common are not provided, no error is thrown. obj.type has to be there and has to be valid.
+ *
+ * @param {object} obj an object which will be validated
+ * @throws Error if a property has the wrong type or obj.type is non existing
+ */
+function validateGeneralObjectProperties(obj) {
+    if (!obj || obj.type === undefined) {
+        throw new Error(`obj.type has to exist`);
+    }
+
+    if (typeof obj.type !== 'string') {
+        throw new Error(`obj.type has an invalid type! Expected "string", received  "${typeof obj.type}"`);
+    }
+
+    const allowedObjectTypes = ['state', 'channel', 'device', 'enum', 'host', 'adapter', 'instance', 'meta', 'config', 'script', 'user', 'group', 'chart'];
+    if (!allowedObjectTypes.includes(obj.type)) {
+        throw new Error(`obj.type has an invalid value (${obj.type}) but has to be one of ${allowedObjectTypes.join(', ')}`);
+    }
+
+    // obj.common is optional in general check
+    if (!obj.common) return;
+
+    if (obj.common.name !== undefined && typeof obj.common.name !== 'string') {
```

Comment:
Name can be an object with many languages: `{"en": "Lamp", "de": "Lampe"}`

Reply:
true, if an object is provided, is attribute 'en' mandatory?